src/coreclr/jit/fginline.cpp

```diff
@@ -613,9 +615,11 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
                 if (!call->IsVirtual())
                 {
                     assert(context != nullptr);
-                    CORINFO_CALL_INFO callInfo = {};
-                    callInfo.hMethod           = method;
-                    callInfo.methodFlags       = methodFlags;
+                    assert(inlinersContext != nullptr);
+                    m_compiler->compInlineContext = inlinersContext;
+                    CORINFO_CALL_INFO callInfo    = {};
+                    callInfo.hMethod              = method;
+                    callInfo.methodFlags          = methodFlags;
```

Comment:
Why do we need to set `exactContextHandle` below? Doesn't `impMarkInlineCandidate``  already set it?

Reply:
It's redundant, removing it now.

Reply:
Arguably we should reset `m_compiler->compInlineContext` back to what it was after we call `impMarkInlineCandidate`.

Or refactor so the context is explicitly passed to `impMarkInlineCandidate` and doesn't need to be a mutable part of the compiler's ambient state.

Reply:
> Or refactor so the context is explicitly passed to `impMarkInlineCandidate` and doesn't need to be a mutable part of the compiler's ambient state.

Agreed with the refactor. Can I refactor `impMarkInlineCandidate` in a following PR? Otherwise we will have to wait another superpmi-asmdiffs-checked-release long run.

Reply:
I always prefer to do the no-diff refactors first and build the diff changes on top. Otherwise it's hard to be sure if the diffs are from the refactoring or the other parts of the change.